### ISSUE TITLE
Support WiFi hardware capabilities

### ISSUE THREAD

**Issue by rgaudin** [offspot/runtime-config#2]
Various RPIs bundles different WiFi capabilities. At least for the models with on-chip WiFi, it would be good to detect the model and adjust the WiFi (AP configuration mostly) settings to make use of those features.
For other models, most likely relying on external WiFi (do we still have such scenarios in the field?), we may imagine some opt-in options in the YAML file.

| Pi | WiFi |
| --- | --- |
| [1A+](https://www.raspberrypi.com/products/raspberry-pi-1-model-a-plus/) | - |
| [1B+](https://www.raspberrypi.com/products/raspberry-pi-1-model-b-plus/) | - |
| [2B](https://www.raspberrypi.com/products/raspberry-pi-2-model-b/) | - |
| [3A+](https://www.raspberrypi.com/products/raspberry-pi-3-model-a-plus/) | 2.4GHz and 5GHz IEEE 802.11.b/g/n/ac wireless LAN |
| [3B](https://www.raspberrypi.com/products/raspberry-pi-3-model-b/) | BCM43438 wireless LAN |
| [3B+](https://www.raspberrypi.com/products/raspberry-pi-3-model-b-plus/) | 2.4GHz and 5GHz IEEE 802.11.b/g/n/ac wireless LAN |
| [pi4-B](https://www.raspberrypi.com/products/raspberry-pi-4-model-b/specifications/) | 2.4 GHz and 5.0 GHz IEEE 802.11ac wireless |
| [pi400](https://www.raspberrypi.com/products/raspberry-pi-400/specifications/) | Dual-band (2.4GHz and 5.0GHz) IEEE 802.11b/g/n/ac wireless LAN |
| [Pico W/WH](https://www.raspberrypi.com/products/raspberry-pi-pico/) | 2.4GHz 802.11n wireless LAN |
| [Zero](https://www.raspberrypi.com/products/raspberry-pi-zero/) | - |
| [Zero W](https://www.raspberrypi.com/products/raspberry-pi-zero-w/) | 802.11 b/g/n wireless LAN |
| [Zero 2 W](https://www.raspberrypi.com/products/raspberry-pi-zero-2-w/) | 2.4GHz 802.11 b/g/n wireless LAN |

**Comment by rgaudin** [comment]
We've reduced the list of supported hardware to Pi `3B` `3B+` `4B` `400` `Zero2W`

**Comment by kelson42** [comment]
So I guess we can close this ticket?

**Comment by rgaudin** [comment]
Nope! It just limits the scope of what we should be able to support. We still only provide 802.11g
